### ISSUE TITLE
Category Page Filter by Product Attributes

### ISSUE THREAD

**Issue by fetullahyldz** [simplcommerce/SimplCommerce#1002]
Hello, I am filtering according to the product color and size, but only one product comes. Please can anyone help me reply?

        var variations = _productRepository
              .Query()
              .Include(x => x.OptionCombinations).ThenInclude(o => o.Option)
              .Include(p => p.LinkedProductLinks)
              .Include(x => x.Medias).ThenInclude(m => m.Media)
              .Where(x => x.IsPublished)
              .ToList();

            var optionsvariant = searchOption.GetOptions();
            if (optionsvariant.Any())
            {
                foreach (var getValue in variations)
                {
                    foreach (var item in getValue.OptionCombinations.Where(x => optionsvariant.Contains(x.Value)))
                    {

                        foreach (var value in getValue.LinkedProductLinks.Where(x => x.LinkedProductId == item.ProductId))
                        {
                            query = query.Where(x => x.Id == value.ProductId);
                        }
                    }
                }

            }

**Comment by fetullahyldz** [comment]
I guess no one is answering here, as if the topics you opened don't matter.

**Comment by nqtuyen** [comment]
according to I see currently filtering by Sub categories, Brand and Price.

**Comment by fetullahyldz** [comment]
Yes, I know this, but it should be done according to the product characteristics, for example, according to the product color and size.
